tests/basic/storage_pool_test.go

```diff
@@ -2054,3 +2423,2175 @@ var _ = Describe("{VolUpdateAddDrive}", func() {
+	//step2: feed p1 size GB I/O on the volume
+	//step3: After I/O done p1 should be go offline and full, expand the pool p1 when p1 is rebalancing add a new drive with different size so that a new pool would be created
+	//step4: validate the pool and the data
+
+	var testrailID = 51443
+	// testrailID corresponds to: https://portworx.testrail.net/index.php?/cases/view/51443
+	var runID int
+
+	JustBeforeEach(func() {
+		StartTorpedoTest("AddNewPoolWhileFullPoolExpanding", "Feed a pool full, then expand the pool when it is rebalancing add another pool", nil, testrailID)
+		runID = testrailuttils.AddRunsToMilestone(testrailID)
+	})
+
+	var contexts []*scheduler.Context
+
+	stepLog := "Create vols and make pool full"
+	It(stepLog, func() {
+		log.InfoD(stepLog)
+		selectedPool := getPoolWithLeastSize()
+		selectedNode, err := GetNodeWithGivenPoolID(selectedPool.Uuid)
+		log.FailOnError(err, fmt.Sprintf("Failed to get node with pool UUID %s", selectedPool.Uuid))
+		log.Infof(fmt.Sprintf("Node %s with pool %s is marked for repl 1", selectedNode.Name, selectedPool.Uuid))
+		stNodes := node.GetStorageNodes()
+		var secondReplNode node.Node
+		for _, stNode := range stNodes {
+			if stNode.Name != selectedNode.Name {
+				secondReplNode = stNode
+			}
+		}
+		secondNodePools := secondReplNode.Pools
+		secondNodePool := secondNodePools[0]
+		log.Infof(fmt.Sprintf("Node %s with pool %s is marked for repl 2", secondReplNode.Name, secondNodePool.Uuid))
+
+		expectPoolSize := selectedPool.TotalSize * 2 / units.GiB
+		isjournal, err := isJournalEnabled()
+		log.FailOnError(err, "is journal enabled check failed")
+		if secondNodePool.TotalSize/units.GiB < expectPoolSize-3 {
+			log.InfoD("Current Size of the pool %s is %d", secondNodePool.Uuid, secondNodePool.TotalSize/units.GiB)
+			err = Inst().V.ExpandPool(secondNodePool.Uuid, api.SdkStoragePool_RESIZE_TYPE_RESIZE_DISK, expectPoolSize)
+			dash.VerifyFatal(err, nil, "Pool expansion init successful?")
+
+			err = waitForPoolToBeResized(expectPoolSize, secondNodePool.Uuid, isjournal)
+			log.FailOnError(err, fmt.Sprintf("Error waiting for poor %s resize", secondNodePool.Uuid))
+		}
+
+		defer Inst().S.RemoveLabelOnNode(*selectedNode, k8s.NodeType)
+		defer Inst().S.RemoveLabelOnNode(secondReplNode, k8s.NodeType)
+		err = Inst().S.AddLabelOnNode(*selectedNode, k8s.NodeType, k8s.FastpathNodeType)
+		log.FailOnError(err, fmt.Sprintf("Failed add label on node %s", selectedNode.Name))
+		err = Inst().S.AddLabelOnNode(secondReplNode, k8s.NodeType, k8s.FastpathNodeType)
+		log.FailOnError(err, fmt.Sprintf("Failed add label on node %s", secondReplNode.Name))
+
+		Inst().AppList = append(Inst().AppList, "fio-fastpath")
+		contexts = make([]*scheduler.Context, 0)
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			contexts = append(contexts, ScheduleApplications(fmt.Sprintf("nwplfullad-%d", i))...)
+		}
+		//ValidateApplications(contexts)
```

Comment:
remove `ValidateApplications(contexts)` if you not using it here

Reply:
fixed